src/test/steps/loginSteps.ts

```diff
@@ -1,37 +1,51 @@
 import {Given, When, Then} from '@cucumber/cucumber';
-import {chromium, Page, Browser, expect} from '@playwright/test';
+import {expect} from '@playwright/test';
+import basePage from '../../pages/basePage';
 
-let browser: Browser;
-let page: Page;
+Given('User passes the authorization', async function () {
+  await basePage.authPage.authorizeUser();
 
-Given('User navigates to the application', async function () {
-  browser = await chromium.launch({headless: false});
-  page = await browser.newPage();
-  page.goto('https://px-stage.self-service.danads.com');
+  const pageTitle = await basePage.page.title();
+  expect(pageTitle).toContain('Core Ad Manager');
 });
 
-Given('User clicks login button', async function () {
-  //await page.locator("//a/*[text()='Log in']").click();
+Given('User clicks Log in button', async function () {
+  await basePage.wrapper.waitAndClick(basePage.loginPage.logInBtn);
 });
 
-Given('User enters username', async function () {
-  await page.locator('(//input[@name=\'username\'])[2]').fill('Username');
+When('User enters email as a {string}', async function (userRole:string) {
+  let email: string;
+
+  switch (userRole) {
+    case 'admin':
+      email = process.env.ADMIN_EMAIL;
+      break;
+    case 'publisher':
+      email = process.env.PUBLISHER_EMAIL;
+      break;
+    default:
+      throw new Error(`Unsupported user role: ${userRole}`);
+  }
```

Comment:
This logic should be moved to a lower-level class since the implementation is not related to business logic.

Reply:
Fixed